symengine/lib/symengine_wrapper.pyx

```diff
@@ -2078,22 +2061,108 @@ class DenseMatrixIter(object):
 
     next = __next__
 
-Matrix = DenseMatrix
+cdef class MutableDenseMatrix(DenseMatrixBase):
+
+    def row_join(self, rhs):
+        cdef DenseMatrixBase o = _sympify(rhs)
+        if self.rows != o.rows:
+            raise ShapeError("`self` and `rhs` must have the same number of rows.")
+        cdef DenseMatrixBase result = zeros(self.rows, self.cols + o.cols)
+        for i in range(self.rows):
+            for j in range(self.cols):
+                result[i, j] = self[i, j]
+        for i in range(o.rows):
+            for j in range(o.cols):
+                result[i, j + self.cols] = o[i, j]
+        return result
+
+    def col_join(self, bott):
+        cdef DenseMatrixBase o = _sympify(bott)
+        if self.cols != o.cols:
+            raise ShapeError("`self` and `rhs` must have the same number of columns.")
+        cdef DenseMatrixBase result = zeros(self.rows + o.rows, self.cols)
+        for i in range(self.rows):
+            for j in range(self.cols):
+                result[i, j] = self[i, j]
+        for i in range(o.rows):
+            for j in range(o.cols):
+                result[i + self.rows, j] = o[i, j]
+        return result
+
+    def col_del(self, i):
```

Comment:
Since these methods are added for SymPy compatibility, having this function do the wrong thing instead is worse than not having this function at all.
Also, you can change the internal pointer `thisptr` to make an inplace modification of the python object, even if you can't change the pointed object.